x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/optimizer/LogicalVerifier.java

```diff
@@ -15,20 +15,22 @@
 
 public final class LogicalVerifier extends PostOptimizationPhasePlanVerifier<LogicalPlan> {
 
-    public static final LogicalVerifier INSTANCE = new LogicalVerifier();
+    private LogicalVerifier(boolean isLocal) {
+        super(isLocal);
+    }
 
-    private LogicalVerifier() {}
+    public static LogicalVerifier getLocalVerifier() {
+        return new LogicalVerifier(true);
```

Comment:
nit: instead of re-instantiating this all the time, we could hand out a singleton. Not sure it makes a difference, though.

We could also make this methods on PostOptimizationPhasePlanVerifier as they work the same for the logical and physical verifiers.

Reply:
> instead of re-instantiating this all the time, we could hand out a singleton.

+1. I guess we'll want this for both flavours (logical, physical).

> Not sure it makes a difference, though.

Maybe not performance-wise, but still has a "better feel". Besides, a node can play both roles, so both types (local/coordinator) would eventually be used.

Reply:
Not sure why the variable which it gets assigned to is not `static` anyway? Probably should make it so?